### ISSUE TITLE
Feature request: Use logr instead of uber/zap directly

### ISSUE THREAD

**Issue by tamalsaha** [cloudevents/sdk-go#690]
We are using cloudevents SDK from our k8s projects. It would be nice if we could just use on logging library. What do you think about using https://github.com/go-logr/logr as the logger. Then users can plugin zap (https://github.com/go-logr/zapr) or klog (https://github.com/kubernetes/klog/tree/main/klogr) or anything else they like to use as the underlying logging library.

**Comment by n3wscott** [comment]
I like the idea!

**Comment by batazor** [comment]
I think it's become even more relevant with the release of the slog.

https://go.dev/blog/slog

**Comment by embano1** [comment]
Using `slog` would require all users (beginning with this release) to use Go 1.21. Not sure if we should enforce this?

**Comment by embano1** [comment]
@brigitops yes, I totally understand this ask. Currently a bit busy and don't have a lot of cycles to go through the code base and see how/if a breaking change can be avoided. Do you have some ideas for a PR that you want to propose?
